repose-aggregator/core/repose-core/src/main/scala/org/openrepose/powerfilter/ReposeRoutingServlet.scala

```diff
@@ -0,0 +1,380 @@
+      .map(_.asInstanceOf[List[RouteDestination]])
+      .getOrElse(List.empty)
+    val defaultDestinations = defaultDestination
+      .map(defDest => new RouteDestination(defDest.getId, servletRequest.getRequestURI, -1))
+      .toList
+    (attributeDestinations ++ defaultDestinations)
+      .sortWith(_.compareTo(_) > 0) // Highest Quality first
+      .headOption match {
+      case Some(routeDest) =>
+        localNode match {
+          case Some(node) => doDestination(servletRequest, servletResponse, routeDest, node)
+          case _ =>
+            logger.error("Invalid local node definition")
+            servletResponse.setStatus(SC_SERVICE_UNAVAILABLE)
+        }
+      case _ =>
+        logger.warn("No Valid routing destination specified.")
+        servletResponse.setStatus(SC_INTERNAL_SERVER_ERROR)
+    }
+  }
+
+  def doDestination(servletRequest: HttpServletRequestWrapper, servletResponse: HttpServletResponse, routeDest: RouteDestination, localhost: Node): Unit = {
+    destinations.get(routeDest.getDestinationId) match {
+      case Some(destination) =>
+        getDestinationLocation(servletRequest, destination, localhost, routeDest.getUri, servletRequest) match {
+          case Some(locationUrl) =>
+            // According to the Java 6 javadocs the routeDestination passed into getContext:
+            // "The given path [routeDestination] must begin with /, is interpreted relative to the server's document root
+            // and is matched against the context roots of other web applications hosted on this container."
+            Option(getServletContext.getContext(locationUrl.getPath)).foreach { targetContext =>
+              // Capture this for Location header processing
+              val locationUrlPath = locationUrl.getPath
+              val targetContextPath = targetContext.getContextPath
+              val dispatchPath = if (locationUrlPath.startsWith(targetContextPath)) locationUrlPath.substring(targetContextPath.length) else locationUrlPath
+              Option(targetContext.getRequestDispatcher(dispatchPath)) match {
+                case Some(dispatcher) =>
+                  doDispatch(
+                    servletRequest,
+                    servletResponse,
+                    locationUrl,
+                    dispatchPath,
+                    targetContext,
+                    routeDest,
+                    dispatcher,
+                    destination)
+                case _ =>
+                  logger.error(s"No Request Dispatcher found in Servlet Context for $dispatchPath; not forwarding")
+              }
+            }
+          case _ =>
+            logger.warn("No route-able node found; not forwarding")
+        }
+      case id =>
+        val localId = localCluster.map(_.getId) match {
+          case Some(locId) => locId
+          case None => ""
+        }
+        logger.warn("Invalid routing destination specified: {} for cluster: {} ", id, localId)
```

Comment:
Huh? `id` must be the `None` object, right? That's probably not what you mean to log. This case should probably be the default case:

``` 
case _ =>
```

... and this log line should log `routeDest.getDestinationId`.

Reply:
Yep that was a bleed over from all the copy/paste refactor and auto convert from Java to Scala.
Updated.